src/iterators.jl

```diff
@@ -309,6 +339,25 @@ function FacetIterator(
     return FacetIterator(FacetCache(gridordh, flags), set)
 end
 
+function FacetIterator(
+        gridordh::Union{<:AbstractGrid, <:DofHandler}, topology::ExclusiveTopology,
+        flags::UpdateFlags = UpdateFlags()
+    )
+    grid = gridordh isa DofHandler ? get_grid(gridordh) : gridordh
+    set = Set(create_boundaryfacetset(grid, topology, _ -> true))
```

Comment:
I'm using sets because ordered sets can allocate while iterating for some reason
And they do something a bit strange I'll have to check it again to make sure it's actually happening not just me hallucinating
The allocate on x86 but not on Arm :D

Reply:
cc @fredrikekre any idea what is going on here?

Reply:
I believe they rehash (if necessary) before iteration.